replay.pokemonshowdown.com/src/replays-battle.tsx

```diff
@@ -159,17 +159,22 @@ export class BattlePanel extends preact.Component<{ id: string }> {
 			this.autofocusTurnView = null;
 		}
 	}
+	// @ts-expect-error really wish they let me assert that the target is an HTMLElement
 	keyPressed = (e: KeyboardEvent) => {
-		this.lastUsedKeyCode = `${e.keyCode}`;
+		this.lastUsedKeyCode = `${e.code}`;
 		if (e.ctrlKey || e.metaKey || e.altKey) return;
-		if (e.keyCode === 27 && this.turnView) { // Esc
+
+		const target = e.target as HTMLElement;
+		if (target.tagName === 'INPUT' || target.tagName === 'SELECT' || target.tagName === 'TEXTAREA' || target.isContentEditable) return;
+
+		if (e.key === 'Escape' && this.turnView) {
 			this.closeTurn();
 			return;
 		}
-		// @ts-expect-error really wish they let me assert that the target is an HTMLElement
-		if (e.target?.tagName === 'INPUT' || e.target?.tagName === 'SELECT') return;
-		switch (e.keyCode) {
-		case 75: // k
+
+		switch (e.key) {
```

Comment:
https://caniuse.com/?search=KeyboardEvent.key

Actually doesn't support _any_ Edge version before 79. Relatively bad compatibility in general.